department-of-veterans-affairs/abd-vro
# Mcp 2009 - update BIP API Service

Update BIP Claims and Evidence API services.

## What was the problem?
The current BIP API service code is not used in the application due to the conflict with the testing. As we updated the environment based building, the conflict is resolved. So this PR enacts the BIP API service. 

BIP Controller is provided for testing some BIP api services. At present, there are five endpoints. The controller used to work with a mocking service in VRO code.  This PR changes it.

1. It is using this mocking service when the app is build in local (default) environment. 
2. It is using BIP UAT service, when it is in dev environment. To build a dev environment, set environment variable ENV=dev. 

Note: 3 out of the five endpoints in BIP Controller are not working at present. 

Associated tickets or Slack threads:
MCP-2009

## How does this fix it?[^1]


## How to test this PR
- Step 1. Follow the normal process to build the app.
- Step 2. Go to the swagger page, test get claim and get contention endpoints.
- Step 3. Set ENV to dev, export ENV=dev. Rebuild and launch the application.
- Step 4. Repeat step 2, using claim ID, 9729039.
- Step 5. run source ./setuatenv.sh (get setuatenv.sh from secret repo). Rebuild and launch the application.
- Step 6. Repeat step 4, you'll see "Name does not resolve" error as you are out side VA network.
Note: Step 4 and 5 should be skipped in latest update. Per Afsin, BIP mocker service will not be used.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
